app/controllers/krikri/records_controller.rb

```diff
@@ -80,5 +80,17 @@ class RecordsController < CatalogController
 
       config.solr_document_model = Krikri::SearchIndexDocument
     end
+
+    ##
+    # Construct a valid item URI from a local name, and use it to fetch a single
+    # document from the search index.
+    # Override method in Blacklight::SolrHelper.
+    # This method is depreciated in Blacklight v5.10.
+    # @param String id is a local name.
+    def get_solr_response_for_doc_id(id=nil, extra_controller_params={})
```

Comment:
Awesome. I'll build this locally and try to merge in time to sneak it into the next Heidrun release.